lib/Driver.js

```diff
@@ -83,80 +84,81 @@ module.exports = class Driver {
    * @returns {Promise<DidDocument>}
    */
   async generate({keyType = 'Ed25519VerificationKey2018'} = {}) {
-    const publicKey = await LDKeyPair.generate({type: keyType});
-    return this.keyToDidDoc(publicKey);
+    const keyPair = await cryptoLd.generate({type: keyType});
+    return this.keyPairToDidDocument({keyPair});
   }
 
   /**
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
-   * @param {Ed25519KeyPair} edKey
-   * @returns {DidDocument}
+   * @param {Ed25519KeyPair} keyPair - The key pair to use to generate the
+   *   DID Document.
+   * @returns {object} - An object containing the `didDocument` and `keyPairs`.
    */
-  async keyToDidDoc(edKey) {
-    const did = `did:key:${edKey.fingerprint()}`;
-    const keyId = `${did}#${edKey.fingerprint()}`;
-    edKey.controller = did;
+  async keyPairToDidDocument({keyPair} = {}) {
+    const did = `did:key:${keyPair.fingerprint()}`;
+    const edKeyPair = await cryptoLd.from(keyPair);
+    edKeyPair.id = `${did}#${edKeyPair.fingerprint()}`;
+    edKeyPair.controller = did;
+
+    const dhKeyPair = await X25519KeyPair.fromEdKeyPair(edKeyPair);
+    dhKeyPair.id = `${did}#${dhKeyPair.fingerprint()}`;
+    dhKeyPair.controller = did;
 
-    const dhKey = await X25519KeyPair.fromEdKeyPair(edKey);
-    dhKey.id = `${did}#${dhKey.fingerprint()}`;
+    // get the public components of each keypair
+    const publicEdKey = await edKeyPair.export({publicKey: true});
+    const publicDhKey = await dhKeyPair.export({publicKey: true});
 
-    const didDoc = {
+    // FIXME: remove private key accidentally exported to public key
```

Comment:
We'll want to get this FIXME addressed ... as FIXMEs are now considered non-kosher in "finalized" code (use issues instead and links in the code as needed). FIXMEs are fine if we're still working through something, but it shouldn't get into a release unless there are outside pressures that make it otherwise impossible to avoid.